src/store/products/index.js

```diff
@@ -0,0 +1,49 @@
+import { createSlice } from '@reduxjs/toolkit';
+import fetchProducts from './fetch';
+
+const initialState = {
+  products: [],
+  pagination: null,
+  isLoading: true,
+};
+
+export const categoriesSlice = createSlice({
+  name: 'products',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchProducts.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(fetchProducts.fulfilled, (state, action) => {
+        const data = { ...action.payload };
+
+        const pagination = {
+          activePage: data.page,
+          totalPages: data.total_pages,
+          nextPage: data.next_page,
+          prevPage: data.prev_page,
+          totalResults: data.total_results_size,
+        };
```

Comment:
aqui se puede deconstruir en lugar de crear un objeto nuevo:
```suggestion
        const {
          page,
          total_pages,
          next_page,
          prev_page,
          total_results_size,
        } = action.payload;

        const pagination = {
          activePage: page,
          totalPages: total_pages,
          nextPage: next_page,
          prevPage: prev_page,
          totalResults: total_results_size,
        };
```